processing/src/main/java/io/druid/segment/data/GenericIndexed.java

```diff
@@ -525,15 +362,15 @@ public int size()
       return size;
     }
 
-    protected T _get(ByteBuffer copyValueBuffer, int startOffset, int endOffset)
+    T bufferedIndexedGet(ByteBuffer copyValueBuffer, int startOffset, int endOffset)
     {
       final int size = endOffset - startOffset;
-      if (startOffset == endOffset) {
+      lastReadSize = size;
+      if (size == 0) {
         return null;
       }
+      copyValueBuffer.clear();
```

Comment:
If the limit had been previously set.  I.e. the `GenericIndexed` was built on top of a `ByteBuffer` where `limit != capacity`, then the `clear()` will reset to `capacity` instead of the `limit` that had been previously set.